### PR TITLE
Missing Modals

### PR DESCRIPTION
**Pull request by rao53**

### CODE REVIEW

**Comment by rao53** [comment]
Screen Shot 2021-04-23 at 1 22 48 PM
Screen Shot 2021-04-23 at 11 34 55 AM

**Comment by rao53** [comment]
To be honest, my audit of many of the currently implemented features don't require confirmation modals. The two images above are all I figured were needed. @Conteo-Inc/dev-team  If there are any more, please let me know. It's fairly simple to implement and should only take me a few mins.

**Comment by elp49** [comment]
> To be honest, my audit of many of the currently implemented features don't require confirmation modals. The two images above are all I figured were needed. @Conteo-Inc/dev-team If there are any more, please let me know. It's fairly simple to implement and should only take me a few mins.

a failed signup and the profile cancel edit button are missing modals as well

File: frontend/src/components/LoginForm.tsx [hunk 1]
```diff
@@ -60,9 +62,33 @@ export default function LoginForm({
   const history = useHistory()
 
   const { from } = location.state || { from: { pathname: "/" } }
+  const [isOpen, setOpen] = React.useState<boolean>(false)
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [type, setType] = useState<NotificationType["type"]>("error")
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [message, setMessage] = useState(
+    "Email and/or password entered incorrectly"
+  )
```
Comment (elp49) [inline frontend/src/components/LoginForm.tsx]: why not set the type and message when you catch the error?

**Review by elp49 — DISMISSED** [review]

File: frontend/src/components/LoginForm.tsx (hunk 1 above)
Comment (rao53) [reply frontend/src/components/LoginForm.tsx]: I could.. Lowkey wouldn't make much difference but I'll have that done.

**Comment by rao53** [comment]
> > To be honest, my audit of many of the currently implemented features don't require confirmation modals. The two images above are all I figured were needed. @Conteo-Inc/dev-team If there are any more, please let me know. It's fairly simple to implement and should only take me a few mins.
> 
> a failed signup and the profile cancel edit button are missing modals as well

I'm not sure what a failed signup would necessarily constitute except the sign-up request fails which I can add. I'll look into the profile cancel edit button rn.

File: frontend/src/components/LoginForm.tsx (hunk 1 above)
Comment (mneedleman) [reply frontend/src/components/LoginForm.tsx]: Instead of the eslint disable comments, just replace `const [foo, setFoo] = ...` with `const [foo] = ...`

**Review by mneedleman — CHANGES_REQUESTED** [review]
One tiny suggestion

**Comment by rao53** [comment]
Screen Shot 2021-04-25 at 12 30 13 PM
Screen Shot 2021-04-25 at 12 05 58 PM

File: frontend/src/components/profile/EditableProfileContent.tsx [hunk 2]
```diff
@@ -352,6 +363,13 @@ export default function EditableProfileContent({
                   Cancel
                 </Button>
               </Grid>
+              <AbstractModal
+                isModalOpen={isModalOpen}
+                handleConfirm={handleConfirm}
+                handleCancel={handleClose}
+                title="Confirmation"
+                description={`You cant undo this action`}
```
Comment (mneedleman) [inline frontend/src/components/profile/EditableProfileContent.tsx]: can't

**Review by mneedleman — CHANGES_REQUESTED** [review]

**Review by mneedleman — APPROVED** [review]

File: frontend/src/components/profile/EditableProfileContent.tsx [hunk 3]
```diff
@@ -352,6 +363,13 @@ export default function EditableProfileContent({
                   Cancel
                 </Button>
               </Grid>
+              <AbstractModal
+                isModalOpen={isModalOpen}
+                handleConfirm={handleConfirm}
+                handleCancel={handleClose}
+                title="Confirmation"
+                description={`You can't undo this action`}
```
Comment (elp49) [inline frontend/src/components/profile/EditableProfileContent.tsx:371]: should this say "Would you like to undo your edits?" instead?

**Review by elp49 — CHANGES_REQUESTED** [review]

File: frontend/src/components/profile/EditableProfileContent.tsx (hunk 3 above)
Comment (rao53) [reply frontend/src/components/profile/EditableProfileContent.tsx:371]: So, what happens here is for when after editing, they press the cancel button which clears anything they just added. The pop up happens only for the cancel button. Pressing the save button in my opinion doesn't need a modal...

Comment (elp49) [reply frontend/src/components/profile/EditableProfileContent.tsx:371]: oh ok, I was reading this wrong. Initially I thought it was telling the user that they cannot cancel their edits.

**Review by elp49 — APPROVED** [review]

**Pull request merged**